pallets/claim/src/lib.rs

```diff
@@ -164,89 +180,172 @@ decl_error! {
 		/// The module doesn't set relyer
 		NoRelayer,
 	}
-}
 
-decl_module! {
-	pub struct Module<T: Config> for enum Call where origin: T::Origin {
-		// Errors must be initialized if they are used by the pallet.
-		type Error = Error<T>;
-		fn deposit_event() = default;
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
 
-		#[weight = 0]
-		pub fn change_relayer(origin, new_relayer: T::AccountId) -> dispatch::DispatchResult {
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		#[pallet::weight(0 + T::DbWeight::get().reads_writes(1,1))]
+		pub fn change_relayer(
+			origin: OriginFor<T>,
+			new_relayer: T::AccountId,
+		) -> DispatchResultWithPostInfo {
 			ensure_root(origin)?;
 			let old_relayer = Relayer::<T>::get();
-			ensure!(Some(&new_relayer) != old_relayer.as_ref(), Error::<T>::RelayerNotChanged);
-			Relayer::<T>::put(&new_relayer);
-			Self::deposit_event(RawEvent::RelayerChanged(new_relayer));
-			Ok(())
+			if old_relayer.is_some() {
+				ensure!(
+					Some(&new_relayer) != old_relayer.unwrap().as_ref(),
+					Error::<T>::RelayerNotChanged
+				);
+			}
```

Comment:
Maybe we can use `old_relayer.flatten()` to convert `Option<Option<T>>` to `Option<T>` and then compare as the old code